pvlib/shading.py

```diff
@@ -191,3 +191,117 @@ def sky_diffuse_passias(masking_angle):
        Available at https://www.nrel.gov/docs/fy18osti/67399.pdf
     """
     return 1 - cosd(masking_angle/2)**2
+
+
+def tracker_shaded_fraction(tracker_theta, gcr, projected_solar_zenith,
+                            cross_axis_slope=0):
+    r"""
+    Shade fraction (FS) for trackers with a common angle on an east-west slope.
+
+    Parameters
+    ----------
+    tracker_theta : numeric
+        The tracker rotation angle in degrees from horizontal.
+    gcr : float
+        The ground coverage ratio as a fraction equal to the collector width
+        over the horizontal row-to-row pitch.
+    projected_solar_zenith : numeric
+        Zenith angle in degrees of the solar vector projected into the plane
+        perpendicular to the tracker axes.
```

Comment:
It seems like it would be a lot easier for the user if the parameter was simply ``solar_zenith`` and the calculation to the projected solar zenith could be done behind the scenes (there should be a function for that in pvlib somewhere).

Reply:
Here is the calculation for psz:
```python
sp = pvl.solarposition.get_solarposition(times, latitude, longitude, ...)
proj_zenith = np.degrees(np.arctan2(
    np.sin(np.radians(sp.azimuth))*np.sin(np.radians(sp.apparent_zenith)),
    np.cos(np.radians(sp.apparent_zenith))))
```
it requires at least an additional argument, the solar azimuth, and is not a trivial calculation.

It is already in pvlib here:

https://github.com/pvlib/pvlib-python/blob/275e6718caf7486cb5b7dcf29acd59499ad51f7f/pvlib/tracking.py#L432

which made me discover a typo in the comments here:

https://github.com/pvlib/pvlib-python/blob/275e6718caf7486cb5b7dcf29acd59499ad51f7f/pvlib/tracking.py#L421

should be the (x, z) plane, which is corrected lower down here:

Reply:
I've created https://github.com/pvlib/pvlib-python/issues/1734 to propose a stand alone PSZ function, but in the process, realized that the existing use in `tracking.py` actually calculates the solar vector as observed in the reference frame of the tracker axes, which may be tilted north or south. So this new function would need to be generalized to different reference frames.